bindings/pyroot/pythonizations/python/ROOT/_pythonization/_th1.py

```diff
@@ -23,10 +24,55 @@ def _imul(self, c):
     return self
 
 
-@pythonization('TH1')
+_np_dtype_dict = {
+    "float64": "TH1D",
+    "float32": "TH1F",
+    "int32": "TH1I",
+    "int8": "TH1C",
+    "int16": "TH1S",
+}
+
+
+def FromNumpy(*args, **kwargs):
+    r"""Function to create histogram object from Numpy arrays"""
+    th, npval = _numpy_getter(_np_dtype_dict, *args, **kwargs)
+    th.FillN(npval[0].size, *npval)
+    return th
+
+
+def _GetBin(self, *args):
+    a = _numpy_content(self._Original_GetBin(*args), *args)
+    return a
+
+
+def _GetBinCenter(self, *args):
+    a = _numpy_content(self._Original_GetBinCenter(*args), *args)
+    return a
+
+
+def _GetBinContent(self, *args):
+    a = _numpy_content(self._Original_GetBinContent(*args), *args)
+    return a
+
+
+def _GetBinLowEdge(self, *args):
+    a = _numpy_content(self._Original_GetBinLowEdge(*args), *args)
+    return a
+
+
+python_funcs = [_GetBin, _GetBinCenter, _GetBinContent, _GetBinLowEdge]
+
+
+@pythonization("TH1")
 def pythonize_th1(klass):
     # Parameters:
     # klass: class to be pythonized
 
     # Support hist *= scalar
     klass.__imul__ = _imul
+    setattr(klass, FromNumpy.__name__, FromNumpy)
```

Comment:
```suggestion
    klass.FromNumpy = FromNumpy
```

Why the `setattr` detour here?

Reply:
@guitargeek Wanted to have a loop initially in case of multiple functions